### PR TITLE
Hl 1132 ahjo versionid

### PR DESCRIPTION
**Pull request by rikuke**
## Description :sparkles:
Includes both **[HL-1132](https://helsinkisolutionoffice.atlassian.net/jira/software/c/projects/HL/boards/204?assignee=557058%3A7956213f-efd2-4425-a95c-cf86d7d77b7b&selectedIssue=HL-1133) and [HL-1081](https://helsinkisolutionoffice.atlassian.net/jira/software/c/projects/HL/boards/204?assignee=557058%3A7956213f-efd2-4425-a95c-cf86d7d77b7b&selectedIssue=HL-1081)**
The callback received from AHJO:
```{
  "message": "Success",
  "requestId": "{123e4567-e89b-12d3-a456-426614174000}",
  "caseId": "HEL 2023-999999",
  "caseGuid": "{123e4567-e89b-12d3-a456-426614174000}",
  "records": [
    {
      "fileURI": "https://othersystem.hel.fi/file1.pdf",
      "status": "Success",
      "hashValue": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
      "versionSeriesId": "{123e4567-e89b-12d3-a456-426614174000}"
    },
    {
      "fileURI": "https://othersystem.hel.fi/file2.pdf",
      "status": "Success",
      "hashValue": "deadbeefabbaacdcdeadbeefabbaacdcdeadbeefabbaacdcdeadbeefabbaacdc",
      "versionSeriesId": "{321e4567-e89b-12d3-a456-426614174000}"
    }
  ]
}
```

which is received after a open case request was not saving the VersionSeriesIds of the records/attachments of the sent applications. VersionSeriesId is required when updating a record to match the attachment of an application to a record in AHJO.  
This PR adds two new Attachment attributes,` ahjo_version_series_id` (for the VersionSeriesId)and `ahjo_hash_value` (for storing the SHA256 hash of the attachment at time when it is sent to AHJO), which are used in the callback endpoint to match the attachment and the record and save the VersionSeriesId when the endpoint is called by the callback.

The failure  callback messsage:

```
{
  "message": "Failure",
  "requestId": "{123e4567-e89b-12d3-a456-426614174000}",
  "records": [
    {
      "fileURI": "https://othersystem.hel.fi/file.pdf",
      "status": "Failure",
      "hashValue": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
    },
    {
      "fileURI": "https://othersystem.hel.fi/file2.pdf",
      "status": "Success",
      "hashValue": "deadbeefabbaacdcdeadbeefabbaacdcdeadbeefabbaacdcdeadbeefabbaacdc"
    }
  ]
}
```
is also handled by logging the failing records.

## Testing :alembic:
`pytest applications/tests/ahjo_integration.py::test_ahjo_callback_success`
`pytest applications/tests/ahjo_integration.py::test_ahjo_callback_failure`

## Additional notes :spiral_notepad:

[HL-1132]: https://helsinkisolutionoffice.atlassian.net/browse/HL-1132?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HL-1081]: https://helsinkisolutionoffice.atlassian.net/browse/HL-1081?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2789.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2789.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2789.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2789.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2789.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2789.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2789.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2789.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe handler result is __failed__ for https://helsinkilisa-ui-handler-pr2789.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2789.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2789.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2789.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2789.dev.hel.ninja** :rocket::rocket::rocket:

File: backend/benefit/applications/api/v1/ahjo_integration_views.py [hunk 1]
```diff
@@ -132,27 +151,59 @@
                 additional_information=info,
             )
 
+            return Response({"message": "Callback received"}, status=status.HTTP_200_OK)
+        except AhjoCallbackError as e:
             return Response(
-                {"message": "Callback received"},
-                status=status.HTTP_200_OK,
+                {"error": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
```
[the remark this answers is not in the document]
Comment (rikuke) [reply backend/benefit/applications/api/v1/ahjo_integration_views.py:157]: Let's leave this here for now, so AHJO developers will possibly get more informative errors during testing.

**Review by EmiliaMakelaVincit — APPROVED** [review]

**Pull request merged**